server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -221,14 +221,13 @@ private static NameResolution resolveAliasRolloverNames(Metadata metadata, Index
 
     private static NameResolution resolveDataStreamRolloverNames(Metadata metadata, DataStream dataStream, boolean isFailureStoreRollover) {
         final DataStream.DataStreamIndices dataStreamIndices = dataStream.getDataStreamIndices(isFailureStoreRollover);
-        assert dataStreamIndices.getWriteIndex() != null : "Unable to roll over dataStreamIndices with no indices";
+        assert dataStreamIndices.getIndices().isEmpty() == false || (isFailureStoreRollover && dataStreamIndices.isRolloverOnWrite())
+            : "Unable to roll over dataStreamIndices with no indices";
 
-        final IndexMetadata originalWriteIndex = metadata.index(dataStreamIndices.getWriteIndex());
-        return new NameResolution(
-            originalWriteIndex.getIndex().getName(),
-            null,
-            dataStream.nextWriteIndexAndGeneration(metadata, dataStreamIndices).v1()
-        );
+        final String originalWriteIndex = dataStreamIndices.getIndices().isEmpty() && dataStreamIndices.isRolloverOnWrite()
```

Comment:
Is there a specific reason that we are drawing a line here and not allowing a rollover of an empty failure store unless the `rolloverOnWrite` flag is set true? I agree that we should be validating that this is indeed the failure store we're operating on, but I figure at this point we're already in the rollover code, and it doesn't seem like the `rolloverOnWrite` flag contributes to the logic in a meaningful way. Are we just trying to catch an invalid state? Is this state really an invalid one or just unexpected?

Reply:
Good thinking @jbaiera . +1 on simplifying it. I do not see any red flags on skipping the flag here (yet).